server/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java

```diff
@@ -51,14 +55,20 @@ public void testRepeatCleanupsDontRemove() throws Exception {
         assertTrue(cleanup.hasCleanupInProgress());
 
         logger.info("-->  unblocking master node");
-        unblockNode("test-repo", masterNode);
+        unblockNode("test-repo", internalCluster().getMasterName());
 
         logger.info("-->  wait for cleanup to finish and disappear from cluster state");
         awaitClusterState(state ->
                 state.custom(RepositoryCleanupInProgress.TYPE, RepositoryCleanupInProgress.EMPTY).hasCleanupInProgress() == false);
+
+        try {
+            cleanupFuture.get();
+        } catch (ExecutionException e) {
+            // ignored and expected
```

Comment:
Can't we be more specific about the exception that we expect here?

Reply:
Yea definitely, sorry for the laziness, made it a proper specific assertion now :)

Reply:
No problem! :)